include/smp/lock.h

```diff
@@ -71,14 +71,6 @@ static inline void *sel4_atomic_exchange(void *ptr, bool_t
     while (!try_arch_atomic_exchange_rlx(&big_kernel_lock.head,
                                          (void *) big_kernel_lock.node_owners[cpu].node,
                                          (void **) &prev)) {
-        if (clh_is_ipi_pending(cpu)) {
-            /* we only handle irq_remote_call_ipi here as other type of IPIs
-             * are async and could be delayed. 'handleIPI' may not return
-             * based on value of the 'irqPath'. */
-            handleIPI(CORE_IRQ_TO_IRQT(cpu, irq_remote_call_ipi), irqPath);
-        }
-
-        arch_pause();
```

Comment:
Why also remove `arch_pause()`? That call has nothing to do with handling IPIs.

Reply:
I was essentially trying to revert https://github.com/seL4/seL4/commit/3d2ae69f9cb184c34b6b56d1365e57693e832db4 which also added arch_pause().  Before the commit the x86 variant directly called __atomic_exchange_n and continued to call __atomic_exchange_n in it's implementation of try_arch_atomic_exchange.  Meanwhile, arch_pause is an empty static inline on arm. 

This part of the lock is supposed to have minimal bus contention as it's just coordinating the insertion into the FIFO queue. If the pause is needed for x86 then it should either be handled by the compilers implementation of __atomic_exchange_n, or introducing it should have been in a separate commit explaining why just having __atomic_exchange_n was insufficient.

Reply:
Seem you are aiming at soething like what I was experimenting with in https://github.com/axel-h/seL4/pull/73/

Reply:
> Seem you are aiming at something like what I was experimenting with in [axel-h#73](https://github.com/axel-h/seL4/pull/73)

I'm trying to resolve the issue with ipiStallCoreCallback on MCS.